public/app/core/logs_model.test.ts

```diff
@@ -700,6 +700,61 @@ describe('logSeriesToLogsModel', () => {
 
     expect(logSeriesToLogsModel(logSeries)).toMatchObject(metaData);
   });
+  it('should return correct metaData when some data frames have empty fields', () => {
+    const logSeries: DataFrame[] = [
+      toDataFrame({
+        fields: [
+          {
+            name: 'ts',
+            type: FieldType.time,
+            values: ['1970-01-01T00:00:01Z'],
+          },
+          {
+            name: 'line',
+            type: FieldType.string,
+            values: ['WARN boooo'],
+            labels: {
+              foo: 'bar',
+              baz: '1',
+              level: 'dbug',
+            },
+          },
+          {
+            name: 'id',
+            type: FieldType.string,
+            values: ['0'],
+          },
+        ],
+        refId: 'A',
+        meta: {
+          searchWords: ['test'],
+          limit: 1000,
+          stats: [{ displayName: 'Summary: total bytes processed', value: 97048, unit: 'decbytes' }],
+          custom: { lokiQueryStatKey: 'Summary: total bytes processed' },
+          preferredVisualisationType: 'logs',
+        },
+      }),
+      {
```

Comment:
Why don't we need to use `toDataFrame` here? Was a bit confused and thought we area creating only one dataframe here.

Reply:
yes, done in https://github.com/grafana/grafana/pull/28645/commits/a0b9ed162bc36a9a13cb14c3a3d93564dbaf582f